### PR TITLE
Add user model and migration

### PR DESCRIPTION
**Pull request by brennanholtzclaw**
Add user model, migrations, and some cart functionality

### CODE REVIEW

File: app/controllers/sessions_controller.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+      redirect_to dashboard_path(id: @user.id)
+    else
+      flash.now[:error] = "Invalid. Try Again."
+      render :new
+    end
+  end
+
+  def destroy
+    session.clear
+    redirect_to "/"
```
Comment (erinnachen) [inline app/controllers/sessions_controller.rb]: probably use a path helper here

File: spec/models/user_spec.rb [hunk 2]
```diff
@@ -0,0 +1,6 @@
+require "rails_helper"
+
+RSpec.describe User, type: :model do
+  it { should validate_presence_of :username }
+  it { should validate_presence_of :password }
+end
```
Comment (erinnachen) [inline spec/models/user_spec.rb]: probably should add a uniqueness check on username

File: config/routes.rb [hunk 3]
```diff
@@ -2,5 +2,14 @@
   resources :items, only: [:index]
 
   resources :cart_items, only: [:create, :index]
+
+  get "/", to: "users#index"
```
Comment (erinnachen) [inline config/routes.rb]: this should probably be `root to: "users#index"`

File: app/controllers/sessions_controller.rb (hunk 1 above)
Comment (brennanholtzclaw) [inline app/controllers/sessions_controller.rb]: According to rake routes "/" doesn't have a path helper

Comment (ksk5280) [inline app/controllers/sessions_controller.rb]: I'm going to change the path helper in my branch for the landing page, so we don't need to worry about it here

File: app/views/shared/_navbar.html.erb [hunk 4]
```diff
@@ -1,9 +1,17 @@
+<<<<<<< HEAD
```
Comment (ksk5280) [inline app/views/shared/_navbar.html.erb]: This line should be removed

**Comment by erinnachen** [comment]
Yay. good to go.

**Pull request merged**